### DISCUSSION TITLE
[Question] Is it possible to have lists of LLM-defined length?

### DISCUSSION THREAD

**Discussion by vedantroy** [guidance-ai/guidance#108] [Q&A]
It looks like we can use `geneach` to generate lists:
```
    "items": [{{#geneach 'items' num_iterations=5 join=', '}}"{{gen 'this' temperature=0.7}}"{{/geneach}}]
```

I'm wondering, is there a way for the LLM to specify the number of items?
For example: I'm doing extraction of quotes from a paragraph, and ideally the LLM would keep on extracting quotes until it ran out.

Also, I'm wondering what the `'this'` means after the `gen` command in this example.

**Comment by slundberg** [comment]
Yes. you can use `stop="stop_string"` instead of num_iterations. Then geneach will test at the end of each loop iteration wether the llm wants to generate the stop string or the start of the next loop item.

As for the meaning of `'this'`. When you are inside a loop the `this` variable refers to the current item in the list we are looping over (or generating). The `gen` command takes the name of the variable you should generate as the first argument.
